mobile/lib/ui/settings/TEMP/lock_screen_option.dart

```diff
@@ -0,0 +1,230 @@
+import "package:photos/ui/components/captioned_text_widget.dart";
+import "package:photos/ui/components/divider_widget.dart";
+import "package:photos/ui/components/menu_item_widget/menu_item_widget.dart";
+import "package:photos/ui/components/title_bar_title_widget.dart";
+import "package:photos/ui/components/title_bar_widget.dart";
+import "package:photos/ui/components/toggle_switch_widget.dart";
+import "package:photos/ui/settings/TEMP/lock_screen_option_password.dart";
+import "package:photos/ui/settings/TEMP/lock_screen_option_pin.dart";
+import "package:photos/ui/tools/app_lock.dart";
+
+class LockScreenOption extends StatefulWidget {
+  const LockScreenOption({super.key});
+
+  @override
+  State<LockScreenOption> createState() => _LockScreenOptionState();
+}
+
+class _LockScreenOptionState extends State<LockScreenOption> {
+  final Configuration _configuration = Configuration.instance;
+  bool? appLock;
+  bool isPinEnabled = false;
+  bool isPasswordEnabled = false;
+
+  @override
+  void initState() {
+    isPasswordEnabled = _configuration.isPasswordSet();
+    isPinEnabled = _configuration.isPinSet();
+    appLock = isPinEnabled ||
+        isPasswordEnabled ||
+        _configuration.shouldShowLockScreen();
+    super.initState();
+  }
+
+  Future<void> _deviceLock() async {
+    await _configuration.removePinAndPassword();
+    setState(() {
+      isPasswordEnabled = _configuration.isPasswordSet();
+      isPinEnabled = _configuration.isPinSet();
+    });
+  }
+
+  Future<void> _pinLock() async {
+    final bool result = await Navigator.of(context).push(
+      MaterialPageRoute(
+        builder: (BuildContext context) {
+          return const LockScreenOptionPin();
+        },
+      ),
+    );
+    setState(() {
+      isPasswordEnabled = _configuration.isPasswordSet();
+      isPinEnabled = _configuration.isPinSet();
+      if (result == false) {
+        appLock = appLock;
+      } else {
+        appLock = isPinEnabled ||
+            isPasswordEnabled ||
+            _configuration.shouldShowLockScreen();
+      }
+    });
+  }
+
+  Future<void> _passwordLock() async {
+    final bool result = await Navigator.of(context).push(
+      MaterialPageRoute(
+        builder: (BuildContext context) {
+          return const LockScreenOptionPassword();
+        },
+      ),
+    );
+    setState(() {
+      isPasswordEnabled = _configuration.isPasswordSet();
+      isPinEnabled = _configuration.isPinSet();
+      if (result == false) {
+        appLock = appLock;
+      } else {
+        appLock = isPinEnabled || isPasswordEnabled;
+      }
+    });
+  }
+
+  Future<void> _onToggleSwitch() async {
+    if (appLock == false && !(isPasswordEnabled || isPinEnabled)) {
+      AppLock.of(context)!.setEnabled(!appLock!);
+      await Configuration.instance.setShouldShowLockScreen(!appLock!);
+    } else {
+      AppLock.of(context)!.setEnabled(!appLock!);
+      await Configuration.instance.setShouldShowLockScreen(!appLock!);
```

Comment:
Both if and else block is doing the same thing?

Reply:
`if (appLock == false && !(isPasswordEnabled || isPinEnabled))` 
   
then `Configuration.instance.setShouldShowLockScreen(!appLock)` is set to `true` by doing `!appLockk`
else
`Configuration.instance.setShouldShowLockScreen(!appLock)` is set to `false` by doing `!appLockk`

Reply:
Either the `if` or `else` block will execute. If both blocks are doing the same thing, let's remove the unnecessary `if` and `else` conditions?